### ISSUE TITLE
GMRES doesn't like reflecting boundaries

### ISSUE THREAD

**Issue by kaibozhu** [robertsj/MATLAB-Transport-Pack#1]
nihao,  I have noticed your GMRES doesn't like reflecting boundaries. I read your code and the problem does seem to exist.  I do not know if you've solved the problem that i am interested in . 
first, 'reset(this.d_boundary)' function of  the GMRESIteration code  sets all boundaries to zero. but It does not seem to include the exit flux from the boundary .If we modify this function, it still doesn't seem to work.......  i got stuck in this problem.......

**Comment by robertsj** [comment]
You are absolutely right.  I never extended the linear equations to include the boundary flux as an unknown.  One complication is that the right-hand side of the A*phi = b represents the uncollided flux.  With more than one reflecting surface, one must iterate on the uncollided flux (though convergence is usually quick enough to use simple iteration).  I did handle this in `libdetran`, my C++ and Python code that used what I learned from MATLAB-Transport-Pack; see https://github.com/robertsj/libdetran/tree/dev.

**Comment by kaibozhu** [comment]
thank you very much. when i  extended the linear equations to include the boundary flux as an unknown as you say.  the ‘phi’ of A*phi = b contains two parts of flux and boundary flux. for 1D situation,the boundary flux represents left and right, so size of 'phi' and 'b' are n+2.  Through a simple test, GMRES did work. Thank you once again
